e2e/LongHaul/service/DeviceOperations.cs

```diff
@@ -71,17 +70,23 @@ public async Task InvokeDirectMethodAsync(Logger logger, CancellationToken ct)
                                 (DateTimeOffset.UtcNow - responsePayload.CurrentTimeUtc).TotalSeconds);
                         }
 
-                        logger.Trace($"Response status: {response.Status}, payload:\n\t{JsonConvert.SerializeObject(response.PayloadAsString)}", TraceSeverity.Information);
+                        logger.Trace(
+                            $"Response status: {response.Status}, payload:\n\t{JsonConvert.SerializeObject(response.PayloadAsString)}",
+                            TraceSeverity.Information);
                         break;
                     }
-                    catch (IotHubServiceException ex) when (ex.ErrorCode == IotHubServiceErrorCode.DeviceNotOnline)
+                    catch (IotHubServiceException ex) when (ex.IsTransient is true)
                     {
-                        logger.Trace($"Caught exception invoking direct method.\n{ex}", TraceSeverity.Warning);
+                        logger.Trace(
```

Comment:
Instead of having a single Trace method with a parameter accepting the severity, we could create severity-specific Trace methods. Just an observation.

Reply:
Just for readability